packages/engines/src/__tests__/ensureNeededBinariesExist.test.ts

```diff
@@ -0,0 +1,112 @@
+import { describe, expect, it, vitest } from 'vitest'
+
+import { ensureNeededBinariesExist } from '..'
+
+describe('ensureNeededBinariesExist', () => {
+  describe('clientEngineType = client', () => {
+    it('when previewFeatures = [], should download library query engine', async () => {
+      const download = vitest.fn()
+      await ensureNeededBinariesExist({
+        clientEngineType: 'client',
+        previewFeatures: [],
+        download,
+      })
+      expect(download).toHaveBeenCalledWith(
+        expect.objectContaining({
+          binaries: {
+            'schema-engine': expect.any(String),
+            'libquery-engine': expect.any(String),
```

Comment:
We care about the keys (type of engine to download), not the values (i.e., the paths on disk).